### PR TITLE
[ADD] Docker publish workflow

### PR DESCRIPTION
**Pull request by ravinkohli**
This PR adds a workflow which publishes a docker image to `ghcr.io` (the github container registry) and [dockerhub/automlorg/autopytorch](https://hub.docker.com/repository/docker/automlorg/autopytorch). It also tests importing in the docker image as well as runs the tests there to ensure that nothing is breaking on docker.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

Note that a Pull Request should only contain one of refactoring, new features or documentation changes.
Please separate these changes and send us individual PRs for each.
For more information on how to create a good pull request, please refer to [The anatomy of a perfect pull request](https://medium.com/@hugooodias/the-anatomy-of-a-perfect-pull-request-567382bb6067).

## Checklist:

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

## Description

As described in the general summary, this PR publishes an image whenever there is a push on `master` and `development`. This image is published to the autopytorch [repository](https://hub.docker.com/repository/docker/automlorg/autopytorch) on docker hub as well as to github container registry.
## Motivation and Context

This PR is required as we would like to have official images of the repository on docker hub. As we don't support installing autopytorch on windows and mac, this change would enable users of the OSs to run autopytorch using docker.

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+name: Publish Docker image
+
+on:
+  push:
+    # Push  to `master` or `development`
+    branches:
+      - master
+      - development
+      - add_docker-publish
```
Comment (eddiebergman) [inline .github/workflows/docker-publish.yml:14]: I'd also add `workflow_dispatch:` as a possible option to `on:`. It let's you manually run the workflow.

https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#workflow_dispatch

**Review by eddiebergman — COMMENTED** [review]
I

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (ravinkohli) [reply .github/workflows/docker-publish.yml:14]: Done

**Review by nabenabe0928 — APPROVED** [review]
Hi, thanks for the PR and sorry for the late response.
It looks good to me:)

**Pull request merged**